src/analyses/goto_rw.h

```diff
@@ -108,8 +109,8 @@ class rw_range_sett
 
   const range_domaint &get_ranges(objectst::const_iterator it) const
   {
-    assert(dynamic_cast<range_domaint*>(it->second)!=0);
-    return *static_cast<range_domaint*>(it->second);
+    assert(dynamic_cast<range_domaint*>(it->second.get()));
```

Comment:
While you're modifying the line, can this become `INVARIANT` or possibly one of the others?